.github/workflows/ci.yml

```diff
@@ -0,0 +1,26 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: [ 1.13.x, 1.15.x, 1.18.x ]
```

Comment:
Right, 1.13 was chosen because it was the lowest version which seemed to work with dependencies. We could bump that to 1.15 as that's used in Debian stable & then target go versions 1.15.x, 1.18.x. ?
I assume arch only uses 1.18 (or the next go latest?)

For latest go, I can try to see if something like `go-version: [ 1.18.x, latest ]` would work.

Reply:
By default Arch only supports the latest official release - 1.18 at this point.

Reply:
updated (just got to squash comments once all OK). Unfortunately it is a manual change if the Go version changes, but I think that is fine.

Reply:
squashed the comments, let me know if you have other thoughts.

Reply:
My gut is telling me the `NewMachineWithBackend(auto)` might be non-deterministic, once we get support for multiple backends in the CI. We can worry about that, if it ever happens.

As-is series looks good IMHO.

Reply:
Yeah, this first baby step is really just to get some testing actually started in CI.
The next baby step is to figure out how to get proper tests running with both fakemachine backends ;-)